### PR TITLE
PHOENIX-6696 Drop legacy phoenix-client jar

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

**Comment by virajjasani** [comment]
Doesn't seem relevant to this PR but I see lots of `refCount leaked` errors in `DateTimeIT`

**Comment by virajjasani** [comment]
Otherwise +1 for dropping phoenix-client and making only embedded jar available (target version only 5.2.x)

File: phoenix-assembly/src/build/components/all-common-jars.xml [hunk 1]
```diff
@@ -55,13 +47,5 @@
         <include>phoenix-pherf.jar</include>
       </includes>
     </fileSet>
-    <fileSet>
-      <directory>${project.basedir}/../phoenix-tools/target</directory>
```
Comment (gjacoby126) [inline phoenix-assembly/src/build/components/all-common-jars.xml:59]: I'm curious -- why does this affect phoenix-tools?

**Review by gjacoby126 — COMMENTED** [review]
lgtm but one question about phoenix-tools

File: pom.xml [hunk 2]
```diff
@@ -1519,7 +1494,6 @@
       </activation>
       <modules>
         <module>phoenix-client-parent</module>
-        <module>phoenix-client-parent/phoenix-client</module>
```
Comment (joshelser) [inline pom.xml:1522]: Drop `phoenix-client-parent/phoenix-client` directory and pom.xml too?

File: phoenix-assembly/src/build/components/all-common-jars.xml (hunk 1 above)
Comment (stoty) [reply phoenix-assembly/src/build/components/all-common-jars.xml:59]: We've dropped the phoenix-tools module/artifact some time ago, this is just an (unrelated) cleanup.

File: pom.xml (hunk 2 above)
Comment (stoty) [reply pom.xml:1522]: We could. 
However, I plan to have at least one more variant that that can coexist  hbase-shaded-client (PHOENIX-6053), and it's easier to keep the structure than to remove and re-add it later.
If you feel strongly about it, then we certainly can remove this.

**Comment by stoty** [comment]
> Doesn't seem relevant to this PR but I see lots of `refCount leaked` errors in `DateTimeIT`

Yes, we have a lot of those lately (we get that in our upstream tests).
I haven't been able to look into that, but it would be great to solve those sooner than later.

File: pom.xml (hunk 2 above)
Comment (stoty) [reply pom.xml:1522]: Oh, you mean the directory.
Sure.

Comment (stoty) [reply pom.xml:1522]: Done.

**Pull request closed without merging**